src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -3647,32 +3648,45 @@ export class ResultsViewPageStore {
         {
             await: () => [this.studyToDataQueryFilter],
             invoke: async () => {
-                let sampleIdentifiers: SampleIdentifier[] = [];
-                let sampleListIds: string[] = [];
+                const customSampleListIds = new ComplexKeySet();
+                const customSampleListStudyIds: string[] = [];
+                const sampleListIds: string[] = [];
                 _.each(
                     this.studyToDataQueryFilter.result,
                     (dataQueryFilter: IDataQueryFilter, studyId: string) => {
                         if (dataQueryFilter.sampleIds) {
-                            sampleIdentifiers = sampleIdentifiers.concat(
-                                dataQueryFilter.sampleIds.map(sampleId => ({
-                                    sampleId,
-                                    studyId,
-                                }))
-                            );
+                            dataQueryFilter.sampleIds.forEach(sampleId => {
+                                customSampleListIds.add({ sampleId, studyId });
+                            });
+                            customSampleListStudyIds.push(studyId);
                         } else if (dataQueryFilter.sampleListId) {
                             sampleListIds.push(dataQueryFilter.sampleListId);
                         }
                     }
                 );
-                let promises: Promise<Sample[]>[] = [];
-                if (sampleIdentifiers.length) {
+
+                const promises: Promise<Sample[]>[] = [];
+
+                if (customSampleListStudyIds.length > 0) {
                     promises.push(
-                        client.fetchSamplesUsingPOST({
-                            sampleFilter: {
-                                sampleIdentifiers,
-                            } as SampleFilter,
-                            projection: REQUEST_ARG_ENUM.PROJECTION_DETAILED,
-                        })
+                        client
+                            .fetchSamplesUsingPOST({
+                                sampleFilter: {
+                                    sampleListIds: customSampleListStudyIds.map(
+                                        studyId => `${studyId}_all`
+                                    ),
+                                } as SampleFilter,
+                                projection:
+                                    REQUEST_ARG_ENUM.PROJECTION_DETAILED,
+                            })
+                            .then(samples => {
+                                return samples.filter(s =>
+                                    customSampleListIds.has(s, [
```

Comment:
this could be a fuckload of samples, i take it that .has is pretty optimal?

Reply:
i always forget, why don't these guys have unique ids that we made on server specifically for this reason?  maybe we stopped doing that?

Reply:
they do, it's that the custom list specification is specified by sample id and study id, not by unique id, so we won't have them here for filtering

Reply:
Im going to do benchmarking to see how fast it is compared to a more specific solution for this